tests/sqllogictests/src/client/http_client.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use reqwest::header::HeaderMap;
+use reqwest::header::HeaderValue;
+use reqwest::Client;
+use reqwest::ClientBuilder;
+use sqllogictest::ColumnType;
+use sqllogictest::DBOutput;
+
+use crate::error::Result;
+use crate::util::parser_rows;
+use crate::util::HttpSessionConf;
+
+pub struct HttpClient {
+    pub client: Client,
+    pub session: Option<HttpSessionConf>,
+}
+
+impl HttpClient {
+    pub fn create() -> Result<HttpClient> {
+        let mut header = HeaderMap::new();
+        header.insert(
+            "Content-Type",
+            HeaderValue::from_str("application/json").unwrap(),
+        );
+        header.insert("Accept", HeaderValue::from_str("application/json").unwrap());
+        let client = ClientBuilder::new().default_headers(header).build()?;
+        Ok(HttpClient {
+            client,
+            session: None,
+        })
+    }
+
+    pub async fn query(&mut self, sql: &str) -> Result<DBOutput> {
+        let url = "http://127.0.0.1:8000/v1/query".to_string();
+        let mut query = HashMap::new();
+        query.insert("sql", serde_json::to_value(sql).unwrap());
+        if let Some(session) = &self.session {
+            query.insert("session", serde_json::to_value(session).unwrap());
+        }
+        let mut response = self.response(sql, &url, true).await?;
+        // Set session from response to client
+        // Then client will same session for different queries.
+        self.session = serde_json::from_value(
```

Comment:
How about read the whole content via `response.bytes()` and deserialize as json instead of calling ` response.as_object()` multiple times?

Or maybe just `response.json()`?